### ISSUE TITLE
fix(model): resolve V4.1 layer-0 DSA indexer semantics so the GLM DSA band can start (next frontier after #13219)

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#13227]
## Working spine
Resolve V4.1's layer-0 DSA indexer semantics so the GLM DSA band can start on layer 0 and the native serve reaches token generation.

## Current state
The loader fix `dfcb2a0ff` derives `cfg.IndexerTypes` for arch `deepseek41` from tensor presence. The published vcruz Q2_K artifact ships indexer tensors only on layers 2, 8, 14, 20, 24, 28, 32, 36 and none on layers 0/1, so the derived schedule classifies layer 0 as "shared". The GLM DSA band contract requires a band to begin on a full-indexer layer: internal/model/glm_dsa_session.go:171-173 returns "cannot start at GLM shared-indexer layer 0"; internal/model/partition.go:127 and internal/model/pipeline.go:134 refuse the same cut; internal/model/dsa_index.go:237-259 (dsaIndexShare) returns false when a shared layer precedes any full layer. The serve routes tokenHiddenGLMDsa -> decodeBandGLMDsa(..., lo=0, ...), so the failure is now a legible fail-closed error instead of the previous internal/model/kernel.go:238 panic: a strict improvement that still generates no token.

## Why this is next
This is the single next unbounded step toward the first physical 30/5 hardware-witnessed rung. The loader schedule (fak#13219) is landed; the forward semantics it exposes is the current blocker.

## Parent context
#13219

## Core through-line
Change the V4.1 layer-0 indexer role resolution -> decodeBandGLMDsa band start at layer 0 -> the forward advances past layer 0 to finite logits -> a forward/band-start test plus the physical two-prompt serve.

## Gold-plating boundary
Do not change the GLM-5.2 sharing contract for glm_moe_dsa checkpoints. Do not lower any roofline gate. Do not claim the 30/5 first witness from a software run.

## Done condition / witness
Defect resolved and tests pass.
Witness: go test ./internal/model/ -run 'V41.*Indexer|BandStart' -count=1

## Definition of done
- [ ] V4.1 layer-0 top-k semantics are defined from evidence (the v41_attention.go IndexSourceLayerIDs/CompressRatios role model, or a documented dense-causal layer 0) and implemented
- [ ] A forward/band-start test drives tokenHiddenGLMDsa past layer 0 with the real strided indexer set to finite logits
- [ ] go test ./internal/model/ -count=1 and go test ./internal/ggufload/ -count=1 are green
- [ ] No hardware-witnessed criterion is marked from a software run

## Acceptance gate
All unit tests pass.

## Closure binding
Resolving commit cites this issue and carries (fak model).

## Lane
model

## Likely files
- internal/model/v41_attention.go
- internal/model/glm_dsa_session.go
- internal/model/dsa_index.go
- internal/model/partition.go
- internal/ggufload/deepseek41.go
## Expected steps
4

- Centrality: Core
- P1 Context: advanced - captures the band-start contract once
- P2 Net value: advanced - unblocks the first physical V4.1 token
- P3 Adaptation: advanced - reuses the existing V41AttentionPlan role model
- P4 Operations: preserved - no regression on the serve/lease path

## Work estimate
Estimate: 4 points

## Overall completion contribution
Contribution: 4/40 points

## Completion standard
demo

**Comment by anthony-chaudhary** [comment]
﻿## Landed: V4.1 layer-0 DSA indexer semantics resolved as `dense`

Resolving commit **283336179** (public `anthony-chaudhary/fak`, pushed to `origin/main`, `bb4c68133..283336179`).

**Root cause (evidence-backed).** V4.1's published schedule (`index_source_layer_ids = {2,8,14,20,24,28,32,36}`, `compress_ratios[0:2] = 0`) makes layers 0/1 indexer-less AND source-less. The #13219 loader derived that correctly but labeled those prefix layers `"shared"`. A leading `"shared"` layer is **unsatisfiable** by the schedule's own contract: `dsaIndexShare` refuses it (`dsa_index.go:250`, `current == nil`) and the band contract forbids starting a band on one (`glm_dsa_session.go:171`, `partition.go:127`, `pipeline.go:134`). The native serve failed closed with `cannot start at GLM shared-indexer layer 0`.

**Fix.** New `"dense"` indexer kind: no indexer tensors AND no preceding full layer → attend the **full causal prefix**, exactly the existing `IndexNHeads==0` dense-MLA seam. This matches V4.1's own reference-faithful role model (`v41AttentionPlanFor` gives a pre-source layer `TopKWidth 0`; `V41AttentionCompressedForward` selects every complete causal group when `Idx` is nil).

- `glm_dsa.go` — `glmDsaIndexerKind` recognizes `dense`/`causal`; add `glmDsaIndexerIsDense`, `glmDsaBandStartOK`.
- `glm_dsa_session.go` / `forward.go` — dense selects the full causal prefix (skips the absent-tensor index step).
- `partition.go` / `pipeline.go` — band/stage boundaries accept a dense head.
- `dsa_index.go` — `dsaIndexShare` publishes nothing for dense and clears the share source (a later shared layer still needs a real full layer).
- `quant.go` / `glm_test.go` — dense enumerates/emits no indexer tensors.
- `ggufload/deepseek41.go` — emit `dense` before the first full layer, `shared` after.

**RED → GREEN.** `TestDeepSeek41GGUFIndexerScheduleDerivedFromTensors` now asserts layers 0/1 `dense`; with the old classification it fails `cfg.IndexerTypes[0] = "shared", want "dense"`. New `v41_dense_indexer_test.go` pins: classification + band start (`decodeBandGLMDsa` over an indexer-less layer-0 fixture no longer returns the band-start refusal); a real `Prefill` over `["dense","dense","full","shared"]` reaching **finite logits past layer 0**; `dsaIndexShare` publishing nothing for dense; and the retained fail-closed halves (unknown type still refused).

**Witness (SW-VERIFIED, not HW).**
- `go test ./internal/model/ -count=1` → `ok` (276.9s)
- `go test ./internal/ggufload/ -count=1` → `ok` (31.9s)
- `go vet ./internal/model/ ./internal/ggufload/` → exit 0
- 5-gate companion boundary `--staged` → `PASS (0 violations across all 5 gates)`

**Classification.** Software leaf only. This issue's acceptance gate is "All unit tests pass" and it is met. It does **NOT** mark any `[HW-WITNESSED]` criterion and no token was generated on hardware; the physical strix3 two-prompt serve remains the witness on the parent track (anthony-chaudhary/fak#12640).

**Comment by anthony-chaudhary** [comment]
﻿## Follow-up: two residual gaps from adversarial cross-validation fixed

Commit **0538f51da** (rebased onto `c1c9fb8bf`, pushed to `origin/main`). An independent adversarial audit of `283336179` returned **CONFIRMED-WITH-CAVEATS**; both genuine findings are now fixed:

1. **Batched forward did not clear the share source on a dense layer.** `glmDsaAttnSeqShared` (`forward.go`) left `*sharedTopK` standing after a dense layer, unlike `dsaIndexShare` (which sets `current=nil`) and the decode path. A `{full, dense, shared}` schedule would have let a following shared layer silently reuse a stale full-prefix selection while `dsaIndexShare` rejects the same schedule. Unreachable from the GGUF loader (dense is only emitted before the first full layer), so it is a correctness guard, not a behavior change on any real checkpoint. New regression `TestV41DenseClearsBatchedShareSource` pins full → dense → shared clears the source and the shared layer then refuses.

2. **`glm_dsa_realdims_test.go` fixture gated indexer emission on `!glmDsaIndexerIsShared`** — a dense layer would have received `indexer.*` tensors (the "dense treated as non-shared" pattern). Corrected to `glmDsaIndexerIsFull`.

**Witness.** `go test ./internal/model/ -count=1` → `ok` (209.9s); `go test ./internal/ggufload/ -count=1` → `ok`; `go vet` clean; 5-gate boundary `PASS (0 violations across all 5 gates)`. No `[HW-WITNESSED]` criterion touched; no token generated on hardware.
